extensions.md

```diff
@@ -165,3 +165,24 @@ clangd allows clients to use UTF-8 offsets instead. This allows clients that alw
   - servers that support both UTF-8 and UTF-16 should check whether the client capabilities mentions `"utf-8"` as supported, and use it if so. The selected encoding should be reported in the `InitializeResponse`.
     This allows UTF-8 to be used when the client prefers it.
   - servers that only support UTF-16 can add `offsetEncoding: "utf-16"` or simply not use the extension.
+
+## Code completion scores
+{:.v10}
+
+LSP gives servers limited control over completion display order through the
+`sortText` attribute. Clangd uses several signals such as number of usages to
+ensure "likely" completions are near the top.
+
+However as the user continues to type, editors filter and re-rank the results
+on the client side. This re-ranking should take into account the signals from
+the server, but LSP does not expose them.
+
+**New CompletionItem property**: `score: number`:
+  - The quality of the result, independent of how well it fuzzy-matches the word
+    being completed. (Value is >= 0, higher is better).
```

Comment:
Maybe also mention that the value is a float between 0 and 1?

Reply:
It's not - it's arbitrary.
We (clangd) use a mostly-0-1 scale for the match score. But actually the top value doesn't matter, just that it's "linear" in some sense. And it's hard to quantify what. I think that part's not well-defined enough to mention at all.